### PR TITLE
unit.j: finish Add/Set/GetResourceAmount

### PR DESCRIPTION
**Pull request by Luashine**
Test map: https://github.com/Luashine/wc3-test-maps/tree/master/AddResourceAmount

### CODE REVIEW

File: unit.j [hunk 1]
```diff
@@ -740,26 +740,43 @@ native IssueNeutralTargetOrderById takes player forWhichPlayer,unit neutralStruc
...
-will display negative resource amount, but if some peasant or peon will try to
-gather resources from such a mine, he will bring back 0 gold and the mine will
-be auto-destroyed.
+@param whichUnit Add gold to this gold mine unit.
 
-@param whichUnit The unit who receives the added resource amount.
+@param amount The amount of gold to add to the unit.
 
-@param amount The amount of resources to add to the unit.
+@note See <SetResourceAmount> for edge-case descriptions. Also: SetResourceAmount, GetResourceAmount
```
Comment (lep) [inline unit.j:767]: Just use the backtick syntax and rely on the web-interface to auto link.

File: unit.j [hunk 2]
```diff
@@ -740,26 +740,43 @@ native IssueNeutralTargetOrderById takes player forWhichPlayer,unit neutralStruc
...
 native AddResourceAmount takes unit whichUnit, integer amount returns nothing
 
-native GetResourceAmount takes unit whichUnit returns integer
 
+/**
+Returns the amount of available gold in a gold mine. The amount can be negative, which is practically the same as 0.
+
+@param whichUnit Add gold to this gold mine unit.
 
+@note See <SetResourceAmount> for edge-case descriptions. Also: SetResourceAmount, AddResourceAmount
```
Comment (lep) [inline unit.j:777]: Same here.

File: unit.j [hunk 3]
```diff
@@ -740,26 +740,43 @@ native IssueNeutralTargetOrderById takes player forWhichPlayer,unit neutralStruc
...
+@bug If the final value, after adding a negative amount, will be less than zero, then it
+will display the correct negative amount, but mining won't yield any gold.
+If peasant enters a mine with 0 gold, it's destroyed and he stops next to mine.
+If peasant enters a mine with <0 gold, it's destroyed and he runs back to the castle.
+
+@param whichUnit Change amount of this gold mine unit.
 
+@param amount The new gold amount.
+
+@note See: AddResourceAmount, GetResourceAmount
```
Comment (lep) [inline unit.j:755]: Use backticks for natives and such.

File: unit.j (hunk 1 above)
Comment (Luashine) [reply unit.j]: Thanks, fixed.

What about other places where <httplink> was used? Like https://github.com/lep/jassdoc/blob/master/builtin-types.j#L12

Comment (lep) [reply unit.j]: That's just markdown syntax for links. They're not strictly necessary as most (all?) implementations automatically convert links. I guess it's just another explicit vs implicit argument. Also there is also the `[title](link)` syntax in markdown which could be used as well but that's like rather old documentation from me. If there ever accumulates more doc for the builtin types those ugly long links could be replaced.

**Pull request merged**